src/Microsoft.Diagnostics.EventFlow.Outputs.HttpOutput/HttpOutput.cs

```diff
@@ -0,0 +1,163 @@
+            }
+
+            if (!string.IsNullOrWhiteSpace(userName) && !string.IsNullOrWhiteSpace(password)) 
+            {
+                string httpAuthValue = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", userName, password)));
+                this.httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", httpAuthValue);
+            }
+
+            switch (this.configuration.Format) 
+            {
+                case "text":
+                    if (string.IsNullOrWhiteSpace(this.configuration.HttpContentType))
+                    {
+                        this.configuration.HttpContentType = "text/plain";
+                    }
+                    break;
+
+                case "json":
+                case "json-lines":
+                    if (string.IsNullOrWhiteSpace(this.configuration.HttpContentType))
+                    {
+                        this.configuration.HttpContentType = "application/json";
+                    }
+                    break;
+
+                default:
+                    errorMessage = $"{nameof(configuration)}: unknown Format \"{nameof(this.configuration.Format)}\" specified";
+                    healthReporter.ReportWarning(errorMessage, EventFlowContextIdentifiers.Configuration);
+                    break;
+            }
+        }
+
+        public async Task SendEventsAsync(IReadOnlyCollection<EventData> events, long transmissionSequenceNumber, CancellationToken cancellationToken)
+        {
+            if (events == null || events.Count == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                var payload = new StringBuilder("");
+
+                switch (configuration.Format) 
+                {
+                    case "text":
+                        foreach (EventData evt in events)
+                        {
+                            payload.AppendLine(evt.ToString());
+                        }
+                        break;
+
+                    case "json":
+                        payload.Append(JsonConvert.SerializeObject(events));
+                        break;
+
+                    case "json-lines":
+                        foreach (EventData evt in events)
+                        {
+                            payload.AppendLine(JsonConvert.SerializeObject(evt));
+                        }
+                        break;
+                }
+
+                HttpContent contentPost = new StringContent(payload.ToString(), Encoding.UTF8, configuration.HttpContentType);
+
+                HttpResponseMessage response = await httpClient.PostAsync(new Uri(configuration.ServiceUri), contentPost);
+                response.EnsureSuccessStatusCode();
+            }
+            catch (Exception ex)
+            {
+                this.healthReporter.ReportProblem($"Fail to send events in batch. Error details: {ex.ToString()}");
```

Comment:
Add something to identify which output produced the error (e.g. nameof(HttpOutput)) and what URL was the data sent to.